### PR TITLE
Expose promise interface in models/config module

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

File: lib/models/config.coffee [hunk 1]
```diff
@@ -24,12 +25,13 @@ request = require('resin-request')
 #		console.log(config)
 ###
 exports.getAll = (callback) ->
-	request.request
-		method: 'GET'
-		url: '/config'
-	, (error, response, config) ->
-		return callback(error) if error?
-		return callback(null, config)
+	Promise.fromNode (callback) ->
```
Comment (Page-) [inline lib/models/config.coffee]: It would be nicer to `Promise.promisifyAll(request)` and then use the async versions

Comment (jviotti) [inline lib/models/config.coffee]: @Page- Fixed, can you have another look please?

File: lib/models/config.coffee [hunk 2]
```diff
@@ -24,12 +25,12 @@ request = require('resin-request')
...
 ###
 exports.getAll = (callback) ->
-	request.request
+	request.requestAsync
 		method: 'GET'
 		url: '/config'
-	, (error, response, config) ->
-		return callback(error) if error?
-		return callback(null, config)
+	.spread (response, config) ->
```
Comment (Page-) [inline lib/models/config.coffee]: This can be `.get(1)`

**Comment by Page-** [comment]
LGTM

File: lib/models/config.coffee (hunk 2 above)
Comment (jviotti) [inline lib/models/config.coffee]: Fixed.

**Pull request merged**